udata/core/user/models.py

```diff
@@ -169,6 +169,12 @@ def generate_api_key(self):
     def clear_api_key(self):
         self.apikey = None
 
+    def can_modify(self, datasets):
```

Comment:
In fact, there is the dataset permission module that centralize these rules, better use it.